### PR TITLE
 Use VisualTreeHelper.GetParent instead of a counter to determine Loaded/Unloaded

### PR DESCRIPTION
**Pull request by victorpadureanu**
Fix for #700

### CODE REVIEW

**Comment by shawnhar** [comment]
With this change, 218 of the 1686 tests in winrt.test.internal are failing.

I expect a couple of the tests will need updating to match the new behavior, but also it looks like the unit test mocks need changing to support this newly added IVisualTreeHelperStatics usage.

File: winrt/lib/xaml/BaseControl.h [hunk 1]
```diff
@@ -129,11 +130,17 @@ namespace ABI { namespace Microsoft { namespace Graphics { namespace Canvas { na
 
         ComPtr<ICanvasDevice> m_customDevice;
 
+        enum class LoadAction
+        {
+            kNoOp = 0,
```
Comment (shawnhar) [inline winrt/lib/xaml/BaseControl.h]: nit: remove the k prefix, to match conventions of other Win2D code.

**Comment by victorpadureanu** [comment]
I used both behaviors and if any of them says loaded it will be loaded.

**Comment by shawnhar** [comment]
Using both behaviors side by side isn't ideal, because it means the tests will not be validating the same code paths that are actually used when Win2D is running inside an app.  This makes the tests significantly less valuable.

The approach used by Win2D is to indirect via adapter interfaces whenever Win2D needs to interact with its host environment.  This lets the unit tests provide emulated versions of whatever XAML functionality Win2D depends on, and mock the behavior of the XAML framework so as to exercise all the possible code paths through Win2D itself in the same way that a real XAML environment would.

For BaseControl, this is done using [IBaseControlAdapter](https://github.com/microsoft/Win2D/blob/master/winrt/lib/xaml/BaseControl.h#L28).  That provides abstract definitions of all the places BaseControl needs to interact with things like its window, UI thread dispatcher, DPI changed event notification, etc.  It can be extended to include access to the visual tree helper (or perhaps just the single GetParent method that this change requires).   BaseControl then uses m_adapter for all access to these features of its outside environment.

In normal usage, IBaseControlAdapter is implemented by [BaseControlAdapter](https://github.com/microsoft/Win2D/blob/master/winrt/lib/xaml/BaseControlAdapter.h#L17), which does a trival pass-through of everything to their real XAML implementations (see GetWindowOfCurrentThread for a simple example).

For unit tests, IBaseControlAdapter is implemented by [BaseControlTestAdapter](https://github.com/microsoft/Win2D/blob/master/winrt/test.internal/xaml/BaseControlTestAdapter.h#L14) and its various subclasses, which allow unit tests to artificially inject whatever result they want the control to observe (see for example the BaseControlTestAdapter implementation of SetHasUIThreadAccess).

Hope this explanation helps, and my apologies for not having this architectural information documented properly anywhere :-(

**Comment by shawnhar** [comment]
A different solution for this issue was incorporated into Win2D release 1.25.0, so this PR is no longer needed

**Pull request closed without merging**